### PR TITLE
F/upgrade to plate 2

### PR DESCRIPTION
**Pull request by bishoymelek**
This one is for upgrading Plate (formally: slate-plugins) to the ls version. 

PS: Some files in table, link plugins is just copied pasted from Plate plugins itself.

### CODE REVIEW

File: packages/common/src/components/emoji-picker/emojiPicker.css [hunk 1]
```diff
@@ -1,4 +1,4 @@
-@import '../../../../node_modules/emoji-mart/css/emoji-mart.css';
+/* @import '~emoji-mart/css/emoji-mart.css'; */
```
Comment (bishoymelek) [inline packages/common/src/components/emoji-picker/emojiPicker.css:1]: We don't use Emoji picker in this PR, I will open a new PR for it and will uncomment this line then.

File: packages/core/src/DreifussWysiwygEditor.tsx [hunk 2]
```diff
@@ -154,14 +147,13 @@ export default function DreifussWysiwygEditor(props: EditorProps) {
             <ToolbarLink />
           </Modal>
           <Divider type={DividerType.vertical} />
-          <ToolbarImage icon={<ImageIcon />} />
```
Comment (bishoymelek) [inline packages/core/src/DreifussWysiwygEditor.tsx:157]: removed this as we have it in line 148 which will be uncommented in the image plugin PR

**Review by penk85 — APPROVED** [review]

**Review by camilleory — APPROVED** [review]

**Review by lakeside763 — APPROVED** [review]

File: packages/core/package.json [hunk 3]
```diff
@@ -94,44 +95,41 @@
...
+    "@udecode/plate-toolbar": "^3.1.0",
+    "@udecode/plate-paragraph": "^2.0.0",
     "jsx-runtime": "^1.2.0",
     "lodash": "^4.17.21",
     "slate": "^0.62.0",
     "slate-history": "^0.62.0",
     "slate-hyperscript": "^0.62.0",
-    "slate-react": "^0.62.0"
+    "slate-react": "^0.62.0",
+    "twin.macro": "^2.7.0"
```
Comment (roosnic1) [inline packages/core/package.json:132]: For what do you need this library?

File: packages/core/src/DreifussWysiwygEditor.tsx (hunk 2 above)
Comment (roosnic1) [reply packages/core/src/DreifussWysiwygEditor.tsx:157]: 👍🏾

File: packages/font-color/package.json [hunk 4]
```diff
@@ -30,18 +30,15 @@
     "react": "^17.0.2",
     "react-dom": "^17.0.2",
     "slate": "^0.60.0",
-    "slate-react": "^0.60.0"
+    "slate-react": "^0.60.0",
+    "@udecode/plate-common": "^2.0.0"
```
Comment (roosnic1) [inline packages/font-color/package.json:34]: why is this a peer Dependency?

File: packages/font-color/package.json [hunk 5]
```diff
@@ -30,18 +30,15 @@
...
   },
   "dependencies": {
     "@babel/polyfill": "^7.12.1",
-    "@udecode/slate-plugins-core": "^1.0.0-next.36",
-    "@udecode/slate-plugins-toolbar": "^1.0.0-next.36"
+    "@udecode/plate-core": "^1.0.0"
   },
   "devDependencies": {
-    "@types/react": "^17.0.14",
-    "@types/react-dom": "^17.0.9",
```
Comment (roosnic1) [inline packages/font-color/package.json:42]: why did you remove those?

File: packages/link-ui/package.json [hunk 6]
```diff
@@ -30,25 +30,26 @@
     "react": "^17.0.2",
     "react-dom": "^17.0.2",
     "slate": "^0.60.0",
-    "slate-react": "^0.60.0"
+    "slate-react": "^0.60.0",
+    "slate-history": "^0.62.0",
+    "@udecode/plate-styled-components": "^3.0.1"
```
Comment (roosnic1) [inline packages/link-ui/package.json:35]: why is this a peer dependency?

File: packages/link-ui/package.json [hunk 7]
```diff
@@ -30,25 +30,26 @@
...
     "@dreifuss-wysiwyg-editor/link": "^1.0.2-alpha.1",
-    "@udecode/slate-plugins-core": "^1.0.0-next.36",
-    "@udecode/slate-plugins-toolbar": "^1.0.0-next.36",
-    "@udecode/slate-plugins-ui-fluent": "^1.0.0-next.36"
+    "@udecode/plate-core": "^1.0.0",
+    "styled-components": "^5.3.1"
   },
   "devDependencies": {
     "@types/react": "^17.0.14",
-    "@types/react-dom": "^17.0.9",
```
Comment (roosnic1) [inline packages/link-ui/package.json:45]: why did you remove this?

File: packages/link-ui/package.json [hunk 8]
```diff
@@ -30,25 +30,26 @@
...
     "@types/slate": "^0.47.8",
     "@types/slate-react": "^0.50.1",
     "react": "^17.0.2",
     "react-dom": "^17.0.2",
     "slate": "^0.62.0",
-    "slate-react": "^0.62.0"
+    "slate-react": "^0.62.0",
```
Comment (roosnic1) [inline packages/link-ui/package.json:51]: why is this still here?

File: packages/link/package.json [hunk 9]
```diff
@@ -33,13 +33,12 @@
...
-    "@udecode/slate-plugins-toolbar": "^1.0.0-next.36",
-    "@dreifuss-wysiwyg-editor/common": "1.0.2-alpha.1"
+    "@udecode/plate-core": "^1.0.0",
+    "@dreifuss-wysiwyg-editor/common": "1.0.2-alpha.1",
+    "@udecode/plate-common": "^2.0.0",
+    "@udecode/plate-normalizers": "^2.0.0"
   },
   "devDependencies": {
-    "@types/react": "^17.0.14",
-    "@types/react-dom": "^17.0.9",
```
Comment (roosnic1) [inline packages/link/package.json:42]: why did you remove those?

File: packages/quotation-mark-ui/package.json [hunk 10]
```diff
@@ -36,15 +36,11 @@
...
     "@dreifuss-wysiwyg-editor/common": "^1.0.2-alpha.1",
     "@dreifuss-wysiwyg-editor/quotation-mark": "0.0.0",
-    "@udecode/slate-plugins-core": "^1.0.0-next.36",
-    "@udecode/slate-plugins-toolbar": "^1.0.0-next.36",
-    "@udecode/slate-plugins-ui-fluent": "^1.0.0-next.36"
+    "@udecode/plate-core": "^1.0.0"
   },
   "devDependencies": {
     "@types/react": "^17.0.14",
-    "@types/react-dom": "^17.0.9",
```
Comment (roosnic1) [inline packages/quotation-mark-ui/package.json:45]: why did you remove this?

File: packages/quotation-mark/package.json [hunk 11]
```diff
@@ -30,17 +30,14 @@
     "react": "^17.0.2",
     "react-dom": "^17.0.2",
     "slate": "^0.60.0",
-    "slate-react": "^0.60.0"
+    "slate-react": "^0.60.0",
+    "@udecode/plate-common": "^2.0.0"
```
Comment (roosnic1) [inline packages/quotation-mark/package.json:34]: why a peer dependency?

File: packages/quotation-mark/package.json [hunk 12]
```diff
@@ -30,17 +30,14 @@
...
   },
   "dependencies": {
-    "@udecode/slate-plugins-core": "^1.0.0-next.36",
-    "@udecode/slate-plugins-toolbar": "^1.0.0-next.36"
+    "@udecode/plate-core": "^1.0.0"
   },
   "devDependencies": {
-    "@types/react": "^17.0.14",
-    "@types/react-dom": "^17.0.9",
```
Comment (roosnic1) [inline packages/quotation-mark/package.json:41]: Why did you remove this?

File: packages/table-ui/package.json [hunk 13]
```diff
@@ -30,24 +30,29 @@
     "react": "^17.0.2",
     "react-dom": "^17.0.2",
     "slate": "^0.60.0",
-    "slate-react": "^0.60.0"
+    "slate-react": "^0.60.0",
+    "slate-history": "^0.62.0",
+    "@udecode/plate-common": "^2.0.0",
+    "styled-components": ">=5.0.0",
+    "twin.macro": "^2.7.0"
```
Comment (roosnic1) [inline packages/table-ui/package.json:37]: Why peer dependencies?

File: packages/table-ui/package.json [hunk 14]
```diff
@@ -30,24 +30,29 @@
...
     "@dreifuss-wysiwyg-editor/common": "^1.0.2-alpha.1",
-    "@udecode/slate-plugins-core": "^1.0.0-next.36",
-    "@udecode/slate-plugins-toolbar": "^1.0.0-next.36"
+    "@udecode/plate-core": "^1.0.0",
+    "@udecode/plate-toolbar": "^3.0.1",
+    "@udecode/plate-styled-components": "^3.0.1"
   },
   "devDependencies": {
     "@types/react": "^17.0.14",
-    "@types/react-dom": "^17.0.9",
```
Comment (roosnic1) [inline packages/table-ui/package.json:44]: Why did you remove this?

File: packages/table/package.json [hunk 15]
```diff
@@ -30,19 +30,16 @@
     "react": "^17.0.2",
     "react-dom": "^17.0.2",
     "slate": "^0.60.0",
-    "slate-react": "^0.60.0"
+    "slate-react": "^0.60.0",
+    "@udecode/plate-common": "2.0.0"
```
Comment (roosnic1) [inline packages/table/package.json:34]: Why peer dependency?

File: packages/table/package.json [hunk 16]
```diff
@@ -30,19 +30,16 @@
...
   "dependencies": {
     "@babel/polyfill": "^7.12.1",
-    "@udecode/slate-plugins-common": "1.0.0-next.36",
-    "@udecode/slate-plugins-core": "1.0.0-next.36",
+    "@udecode/plate-core": "1.0.0",
     "@dreifuss-wysiwyg-editor/common": "^1.0.2-alpha.1"
   },
   "devDependencies": {
-    "@types/react": "^17.0.14",
-    "@types/react-dom": "^17.0.9",
```
Comment (roosnic1) [inline packages/table/package.json:43]: why did you remove this?

**Review by roosnic1 — CHANGES_REQUESTED** [review]
Looks good but i think there are still some confusions with the dependencies.

File: packages/common/package.json [hunk 17]
```diff
@@ -30,15 +30,15 @@
     "react": "^17.0.2",
     "react-dom": "^17.0.2",
     "slate": "^0.60.0",
-    "slate-react": "^0.60.0"
+    "slate-react": "^0.60.0",
+    "@udecode/plate-toolbar": "^3.0.1"
```
[the remark this answers is not in the document]
Comment (bishoymelek) [reply packages/common/package.json:34]: @roosnic1 because it's in dependencies in the core package

File: packages/character-count-ui/package.json [hunk 18]
```diff
@@ -30,17 +30,16 @@
...
+    "lodash": "^4.17.21"
   },
   "dependencies": {
-    "@udecode/slate-plugins-core": "^1.0.0-next.36",
+    "@udecode/plate-core": "^1.0.0",
     "@dreifuss-wysiwyg-editor/common": "^1.0.2-alpha.1"
   },
   "devDependencies": {
     "@types/react": "^17.0.14",
-    "@types/react-dom": "^17.0.9",
```
[the remark this answers is not in the document]
Comment (bishoymelek) [reply packages/character-count-ui/package.json:41]: @roosnic1  for all of these typing packages changes, Honestly i am not sure about it. it's defined in core package, and [`depchck` ](https://www.npmjs.com/package/depcheck) sees it as unused. Also when i tried to run and/or build the package, it works/builds as intended.

File: packages/table/package.json (hunk 15 above)
Comment (bishoymelek) [reply packages/table/package.json:34]: we use it in `dependencies` in core package

File: package.json [hunk 19]
```diff
@@ -73,7 +73,10 @@
     "url-loader": "^4.1.1",
     "webpack": "^5.41.1",
     "webpack-cli": "^4.7.2",
-    "webpack-dev-server": "^3.11.2"
+    "webpack-dev-server": "^3.11.2",
+    "babel-plugin-macros": "^3.1.0",
+    "babel-plugin-twin": "^1.0.2",
```
[the remark this answers is not in the document]
Comment (bishoymelek) [reply package.json:78]: @roosnic1 the twin one is for twin package which is being used for styling (tailwind-css) and macros package is to fix something in babel required in its configuration, check [here](https://github.com/kentcdodds/babel-plugin-macros#the-problem)

File: packages/character-count-ui/package.json (hunk 18 above)
[the remark this answers is not in the document]
Comment (roosnic1) [reply packages/character-count-ui/package.json:41]: @bishoymelek if the package is in the `peerDependencies` then the `@types` should be in the `devDependencies`

File: packages/character-count-ui/package.json [hunk 20]
```diff
@@ -30,15 +30,15 @@
     "react": "^17.0.2",
     "react-dom": "^17.0.2",
     "slate": "^0.60.0",
-    "slate-react": "^0.60.0"
+    "slate-react": "^0.60.0",
+    "lodash": "^4.17.21"
```
Comment (roosnic1) [inline packages/character-count-ui/package.json:34]: and i think this should also not be in the `peerDependencies`

File: packages/core/package.json [hunk 21]
```diff
@@ -80,7 +80,8 @@
     "@storybook/addon-actions": "^6.3.6",
     "@storybook/addon-essentials": "^6.3.6",
     "@storybook/addon-links": "^6.3.6",
-    "@storybook/react": "^6.3.6"
+    "@storybook/react": "^6.3.6",
+    "@types/styled-components": "^5.1.14"
```
[the remark this answers is not in the document]
Comment (roosnic1) [reply packages/core/package.json:84]: ???

File: packages/core/package.json (hunk 3 above)
Comment (roosnic1) [reply packages/core/package.json:132]: ????

File: packages/font-color-ui/package.json [hunk 22]
```diff
@@ -30,19 +30,17 @@
...
-    "@udecode/slate-plugins-core": "^1.0.0-next.36",
-    "@udecode/slate-plugins-toolbar": "^1.0.0-next.36",
-    "@dreifuss-wysiwyg-editor/common": "^1.0.2-alpha.1",
-    "@udecode/slate-plugins-ui-fluent": "^1.0.0-next.36"
+    "@udecode/plate-core": "^1.0.0",
+    "@dreifuss-wysiwyg-editor/common": "^1.0.2-alpha.1"
   },
   "devDependencies": {
     "@types/react": "^17.0.14",
-    "@types/react-dom": "^17.0.9",
```
Comment (roosnic1) [inline packages/font-color-ui/package.json:45]: if it's in the  `peerDependencies` then the `@types` are needed here.

File: packages/link-ui/package.json [hunk 23]
```diff
@@ -48,7 +49,7 @@
     "react": "^17.0.2",
     "react-dom": "^17.0.2",
     "slate": "^0.62.0",
-    "slate-react": "^0.62.0"
```
Comment (roosnic1) [inline packages/link-ui/package.json:51]: Pretty sure it's needed

File: packages/quotation-mark-ui/package.json [hunk 24]
```diff
@@ -36,15 +36,12 @@
...
-    "@udecode/slate-plugins-core": "^1.0.0-next.36",
-    "@udecode/slate-plugins-toolbar": "^1.0.0-next.36",
-    "@udecode/slate-plugins-ui-fluent": "^1.0.0-next.36"
+    "@udecode/plate-core": "^1.0.0"
   },
   "devDependencies": {
     "@types/react": "^17.0.14",
     "@types/react-dom": "^17.0.9",
     "@types/slate": "^0.47.8",
-    "@types/slate-react": "^0.50.1",
```
Comment (roosnic1) [inline packages/quotation-mark-ui/package.json:47]: it's needed.

File: packages/quotation-mark/package.json [hunk 25]
```diff
@@ -33,14 +33,13 @@
...
-    "@udecode/slate-plugins-core": "^1.0.0-next.36",
-    "@udecode/slate-plugins-toolbar": "^1.0.0-next.36"
+    "@udecode/plate-core": "^1.0.0",
+    "@udecode/plate-common": "^2.0.0"
   },
   "devDependencies": {
     "@types/react": "^17.0.14",
     "@types/react-dom": "^17.0.9",
     "@types/slate": "^0.47.8",
-    "@types/slate-react": "^0.50.1",
```
Comment (roosnic1) [inline packages/quotation-mark/package.json:43]: needed

File: packages/table-ui/package.json [hunk 26]
```diff
@@ -30,14 +30,19 @@
     "react": "^17.0.2",
     "react-dom": "^17.0.2",
     "slate": "^0.60.0",
-    "slate-react": "^0.60.0"
+    "slate-react": "^0.60.0",
+    "slate-history": "^0.62.0",
+    "@udecode/plate-common": "^2.0.0",
```
Comment (roosnic1) [inline packages/table-ui/package.json:35]: Why is this still here?

File: packages/table-ui/package.json [hunk 27]
```diff
@@ -30,14 +30,19 @@
...
+    "styled-components": ">=5.0.0"
```
Comment (roosnic1) [inline packages/table-ui/package.json:36]: Why is this still here?

File: packages/table-ui/package.json [hunk 28]
```diff
@@ -47,7 +52,7 @@
     "react": "^17.0.2",
     "react-dom": "^17.0.2",
     "slate": "^0.62.0",
-    "slate-react": "^0.62.0"
```
Comment (roosnic1) [inline packages/table-ui/package.json:50]: needed.

File: packages/table/package.json [hunk 29]
```diff
@@ -34,15 +34,14 @@
...
-    "@dreifuss-wysiwyg-editor/common": "^1.0.2-alpha.1"
+    "@udecode/plate-core": "1.0.0",
+    "@dreifuss-wysiwyg-editor/common": "^1.0.2-alpha.1",
+    "@udecode/plate-common": "2.0.0"
   },
   "devDependencies": {
     "@types/react": "^17.0.14",
     "@types/react-dom": "^17.0.9",
     "@types/slate": "^0.47.8",
-    "@types/slate-react": "^0.50.1",
```
Comment (roosnic1) [inline packages/table/package.json:45]: needed.

**Review by roosnic1 — CHANGES_REQUESTED** [review]
still needs a few changes.

File: packages/character-count-ui/package.json (hunk 20 above)
Comment (bishoymelek) [reply packages/character-count-ui/package.json:34]: added it to dependencies

File: packages/table-ui/package.json (hunk 26 above)
Comment (bishoymelek) [reply packages/table-ui/package.json:35]: moved to dependencies

File: packages/quotation-mark/package.json (hunk 25 above)
Comment (bishoymelek) [reply packages/quotation-mark/package.json:43]: fixed

File: packages/table-ui/package.json (hunk 27 above)
Comment (bishoymelek) [reply packages/table-ui/package.json:36]: @roosnic1 What do you suggest? move it to dependencies? or leave it and add it to dependencies in core package?

File: packages/table-ui/package.json (hunk 28 above)
Comment (bishoymelek) [reply packages/table-ui/package.json:50]: fixed

Comment (bishoymelek) [reply packages/table-ui/package.json:50]: @roosnic1 fixed
Is it normal to have these four packages in peerDependencies and devDependencies?

File: packages/table/package.json (hunk 29 above)
Comment (bishoymelek) [reply packages/table/package.json:45]: fixed

File: packages/table-ui/package.json (hunk 27 above)
Comment (roosnic1) [reply packages/table-ui/package.json:35]: If `styled-components` is needed then add it to the dependencies. I don't understand why you would want to add it to the peerDependencies

Comment (bishoymelek) [reply packages/table-ui/package.json:35]: same as slate and react, they are both needed but will always be there, that's why I suggest we add styled-components in core as we use it in multiple plugins and plate talked about adding it [here](https://plate.udecode.io/docs/installation)

File: packages/core/package.json (hunk 21 above)
[the remark this answers is not in the document]
Comment (bishoymelek) [reply packages/core/package.json:84]: @roosnic1 I added the package on line 131 in dependencies

File: packages/table-ui/package.json (hunk 27 above)
Comment (bishoymelek) [reply packages/table-ui/package.json]: resolved 👍

File: packages/table-ui/package.json (hunk 28 above)
Comment (bishoymelek) [reply packages/table-ui/package.json:50]: fixed 👍

File: packages/font-color-ui/package.json (hunk 22 above)
Comment (bishoymelek) [reply packages/font-color-ui/package.json]: fixed

**Review by roosnic1 — APPROVED** [review]
👍🏾
